source/assets/scripts/main.js

```diff
@@ -10,30 +10,124 @@ const recipes = [
-        if (!response.ok) {
-          throw new Error('Network response was not OK');
-        }
-        return response.json();
-      })
-      .then((data) => {
-        makeCard(data);
-      })
-      .catch((error) => {
-        console.error('Fetch failed', error);
-      });
+    // Makes a new recipe card
+    const recipeCard = document.createElement('recipe-card');
+    // Inputs the data for the card. This is just the first recipe in the recipes array,
+    // being used as the key for the recipeData object
+    recipeCard.data = recipeData[recipes[i]];
+
+    // This gets the page name of each of the arrays - which is basically
+    // just the filename minus the .json. Since this is the first element
+    // in our recipes array, the ghostCookies URL, we will receive the .json
+    // for that ghostCookies URL since it's a key in the recipeData object, and
+    // then we'll grab the 'page-name' from it - in this case it will be 'ghostCookies'
+    let page = recipeData[recipes[i]].title;
+    while (page.includes(' ')) {
+      page = page.replace(' ', '-');
+    }
+    router.addPage(page, () => {
+      document.querySelector('.section--main-page').classList.remove('shown');
+      document.querySelector('.section--recipe-display').classList.add('shown');
+      document.querySelector('recipe-display').data = recipeData[recipes[i]];
+    });
+    bindRecipeCard(recipeCard, page);
+    document.querySelector('.card-body').appendChild(recipeCard);
   }
 }
 
-window.addEventListener('DOMContentLoaded', init);
+/**
+ * Binds the click event listener to the <recipe-card> elements added to the page
+ * so that when they are clicked, their card expands into the full recipe view mode
+ * @param {Element} recipeCard the <recipe-card> element you wish to bind the event
+ *                             listeners to
+ * @param {String} pageName the name of the page to navigate to on click
+ */
+function bindRecipeCard(recipeCard, pageName) {
+  recipeCard.addEventListener('click', (e) => {
+    if (e.path[0].nodeName === 'A') return;
+    router.navigate(pageName);
+  });
+}
+
+/**
+ * Binds the 'keydown' event listener to the Escape key (esc) such that when
+ * it is clicked, the home page is returned to
+ */
+function bindEscKey() {
+  document.addEventListener('keydown', (event) => {
+    if ((event.key === 'Escape') || (event.key === 'Esc')) {
+      router.navigate('home', false);
+    }
+  });
+}
+
+/**
+ * Binds the 'popstate' event on the window (which fires when the back &
+ * forward buttons are pressed) so the navigation will continue to work
+ * as expected. (Hint - you should be passing in which page you are on
+ * in your Router when you push your state so you can access that page
+ * info in your popstate function)
+ */
+function bindPopstate() {
+  window.addEventListener('popstate', (event) => {
+    if (event.state) {
+      router.navigate('home', true);
+    }
+    router.navigate(event.state.page, true);
+  });
```

Comment:
If the state exists, don't we navigate to the state page, rather than home? And shouldn't there be an else statement here -- right now it always navigates to event.state.page.